superset/viz.py

```diff
@@ -1055,9 +1055,9 @@ def query_obj(self):
         # dedup groupby if it happens to be the same
         d["groupby"] = list(dict.fromkeys(d["groupby"]))
 
-        self.x_metric = form_data.get("x")
-        self.y_metric = form_data.get("y")
-        self.z_metric = form_data.get("size")
+        self.x_metric = form_data["x"]
```

Comment:
I perceive these are all required otherwise `utils.get_metric_name(...)` etc. would return `None` and cause issues when trying to index the Pandas DataFrame.